src/recipe.js

```diff
@@ -4,13 +4,19 @@ class Recipe {
     this.id = recipe.id;
     this.image = recipe.image;
     this.ingredients = recipe.ingredients;
-    this.tags = recipe.tags;
-  }
-  calculateIngredientsCost() {
-    // return this.ingredients.map(i => {
-    //   ingredientData.find(ingredient => ingredient === i);
-    // });
-  }
+		this.tags = recipe.tags;
+		this.instructions = recipe.instructions;
+	}
+	
+  calculateIngredCost(dataset) {
+		const totalCost = this.ingredients.reduce((acc, ingred) => {
+			const itemCost = dataset.find(i => i.id === ingred.id).estimatedCostInCents;
```

Comment:
I like how you did this calculation.  i was thinking a map then reduce the mapped but i actually think this is less code and less loops.  nice work!

Reply:
thanks! 🕶️